new_platforms/scripts/VirtualMachineSgxSettings.psm1

```diff
@@ -0,0 +1,283 @@
+﻿Import-Module Hyper-V
+
+### -----------------------------------
+### Constants
+### -----------------------------------
+
+$VM_MEMORY_SETTINGS_GUID="4764334d-e001-4176-82ee-5594ec9b530e"
+
+Set-StrictMode -Version 5
+
+$ErrorActionPreference = "Stop"
+$InformationPreference = "Continue"
+
+### -----------------------------------
+### Strings
+### -----------------------------------
+
+Data Strings
+{
+# culture="en-US"
+ConvertFrom-StringData @'
+    ERR_NO_MEM_SETTING_DATA_OBJECT = Could not retreive the Memory Setting Data object for VM '{0}' from WMI.
+    ERR_NO_VMM = Could not retreive the instance of the Virtual System Management Service from WMI.
+    ERR_VM_STATE = "The VM '{0}' must be Off before its SGX settings may be modified."
+    ERR_NO_SETTINGS_TO_MODIFY = Specify at least one SGX setting to modify.
+    ERR_NO_SGX_MEM_TO_ENABLE = To enable SGX, some SGX EPC memory is required (-SgxSize).
+    ERR_COULD_NOT_MODIFY_SETTINGS = Could not modify SGX settings for VM '{0}' via WMI with error value '{1}'.
+
+    WARN_GEN2_TITLE = The VM {0}' is not a Generation 2 or later VM.
```

Comment:
shouldn't there be a matching single quote before {0}?

Reply:
Added.